### PR TITLE
Subsidium

### PR DESCRIPTION
**Pull request by rubetsserhii**
Whitelist support and XBTX support

### CODE REVIEW

File: libs/paymentProcessor.js [hunk 1]
```diff
@@ -69,23 +69,6 @@ function SetupForPool(logger, poolOptions, setupFinished){
...
-                            logger.error(logSystem, logComponent,
-                                    'Daemon does not own pool address - payment processing can not be done with this daemon, '
-                                    + JSON.stringify(result.response));
-                            callback(true);
-                        }
-                        else{
-                            callback()
-                        }
-                    }, true);
-                }
```
Comment (DancingAxolotl) [inline libs/paymentProcessor.js:88]: Why were these checks removed?

File: libs/poolWorker.js [hunk 2]
```diff
@@ -5,6 +5,23 @@ var net     = require('net');
...
+loadWhitelist = function() {
+    var fs = require('fs');
+    JSON.minify = JSON.minify || require("node-json-minify");
+    
+    if (!fs.existsSync('./whitelist.json')) {
+        console.log('whitelist.json file does not exist. All workers would be accepted.');
+        return [];
+    }
+    
+    var whitelistConfig = JSON.parse(JSON.minify(fs.readFileSync("./whitelist.json", {encoding: 'utf8'})));
```
Comment (DancingAxolotl) [inline libs/poolWorker.js]: Is there a reason to minify JSON before parsing? The parser should work the same on minified and regular files.

**Review by DancingAxolotl — APPROVED** [review]

File: libs/paymentProcessor.js (hunk 1 above)
Comment (rubetsserhii) [reply libs/paymentProcessor.js:88]: there is no "getaddressinfo" in BitcoinSubsidium daemon

File: libs/poolWorker.js (hunk 2 above)
Comment (rubetsserhii) [reply libs/poolWorker.js]: You are right. The code can be simplified.

**Pull request merged**